morph/react/get-formatters.js

```diff
@@ -0,0 +1,38 @@
+export default state => (state.hasFormattedChild ? createFormatters(state) : '')
+
+const createFormatters = ({ formats, localSupported }) => {
+  let formatters = ''
+  formats.forEach(format => {
+    const style = Object.keys(format)[0]
+    if (style === 'currency' || style === 'percent') {
+      formatters += getFormat(format, style, localSupported, 'Number')
+    } else {
+      formatters += getFormat(format, style, localSupported, 'DateTime')
+    }
+  })
+  return formatters
+}
```

Comment:
how about mapping over the formats and then returning a joint array like:
```
return formats.map(format => {
  const style = Object.keys(format)[0]
  return getFormat(format, style, localSupported, style === 'currency' || style === 'percent'? 'Number' : 'DateTime')
}).join('')
```

which after doing that actually I think you could leave the `style` and `type` definition to the `getFormat` method and this would then look like:

```
return formats.map(format => getFormat(format, localSupported)).join('')
```

thoughts?